src/webview/handleOutboundEvents.js

```diff
@@ -184,7 +184,13 @@ const fetchMore = (props: Props, event: WebViewOutboundEventScroll) => {
 
 const markRead = (props: Props, event: WebViewOutboundEventScroll) => {
   const { doNotMarkMessagesAsRead, flags, auth } = props.backgroundData;
-  if (doNotMarkMessagesAsRead) {
+  if (doNotMarkMessagesAsRead === 'stream' && !isStreamOrTopicNarrow(props.narrow)) {
+    return;
+  }
+  if (doNotMarkMessagesAsRead === 'topic' && !isTopicNarrow(props.narrow)) {
+    return;
+  }
+  if (doNotMarkMessagesAsRead === 'never') {
```

Comment:
From this implementation, it looks like the meaning of this value is more like "*do* mark messages as read", rather than "do not mark messages as read".

In that case, let's rename it to match its meaning. One good name would be `shouldMarkAsReadOnScroll`.